feat: add functions to yaml

rui-mo opened the pull request:
This PR added starts_with, ends_with, contains functions to yaml files.

jvanstraten commented on extensions/functions_comparison.yaml:
```diff
@@ -73,4 +73,12 @@ scalar_functions:
+    description: Whether a value is in a list of values.
+    impls:
+      - args:
+          - value: any1
```
I think you meant this?

```suggestion
          - value: T
```

rui-mo replied:
The function we need is already covered by OrList expression. Removed this.

jacques-n commented:
I think you're looking for the orlist expression (as opposed to introducing in).

For the character matching functions, I suggest you also include all permutations of fixed character, string and varchar combinations.

rui-mo commented:
> I think you're looking for the orlist expression (as opposed to introducing in).
> 
> For the character matching functions, I suggest you also include all permutations of fixed character, string and varchar combinations.

Thanks for your comment. Removed in function and added the permutations for starts_with, ends_with, contains.

jacques-n commented:
Please also update commit title. Should be feat and more specific

rui-mo commented:
> Please also update commit title. Should be feat and more specific

Updated. Thank you.

jacques-n reviewed (approved):
Looks mostly good. I will update the title to be more explicit when I merge.
